pkg/eks/filecache_test.go

```diff
@@ -0,0 +1,177 @@
+			Expect(value.SessionToken).To(Equal("token"))
+			Expect(p.IsExpired()).NotTo(BeTrue())
+			content, err := ioutil.ReadFile(filepath.Join(tmp, "credentials.yaml"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(string(content)).To(Equal(`profiles:
+  profile:
+    credential:
+      accesskeyid: id
+      secretaccesskey: secret
+      sessiontoken: token
+      providername: stubProvider
+    expiration: 0001-01-01T00:00:00Z
+`))
+			Expect(p.IsExpired()).NotTo(BeTrue())
+		})
+		When("the cache expires", func() {
+			It("will ask to refresh it", func() {
+				c := credentials.NewCredentials(&stubProviderExpirer{
+					stubProvider: stubProvider{
+						creds: credentials.Value{
+							AccessKeyID:     "id",
+							SecretAccessKey: "secret",
+							SessionToken:    "token",
+							ProviderName:    "stubProvider",
+						},
+					},
+				})
+				fakeClock := &fakes.FakeClock{}
+				fakeClock.NowReturns(time.Date(9999, 1, 1, 1, 1, 1, 1, time.UTC))
+				p, err := NewFileCacheProvider("profile", c, fakeClock)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(p.IsExpired()).To(BeTrue())
+			})
+		})
+		When("the cache file already exists", func() {
+			It("will retrieve its content", func() {
+				content := []byte(`profiles:
+  profile:
+    credential:
+      accesskeyid: storedID
+      secretaccesskey: storedSecret
+      sessiontoken: storedToken
+      providername: stubProvider
+    expiration: 0001-01-01T00:00:00Z
+`)
+				err := ioutil.WriteFile(filepath.Join(tmp, "credentials.yaml"), content, 0700)
+				Expect(err).ToNot(HaveOccurred())
+				c := credentials.NewCredentials(&stubProviderExpirer{})
+				fakeClock := &fakes.FakeClock{}
+				p, err := NewFileCacheProvider("profile", c, fakeClock)
+				Expect(err).ToNot(HaveOccurred())
+				creds, err := p.Retrieve()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(creds.AccessKeyID).To(Equal("storedID"))
+				Expect(creds.SecretAccessKey).To(Equal("storedSecret"))
+				Expect(creds.SessionToken).To(Equal("storedToken"))
+			})
+		})
+		When("no underlying credentials have been supplied", func() {
+			It("returns an appropriate error", func() {
+				fakeClock := &fakes.FakeClock{}
+				_, err := NewFileCacheProvider("profile", nil, fakeClock)
+				Expect(err).To(MatchError("no underlying Credentials object provided"))
+			})
+		})
+		When("the underlying credentials provider doesn't support caching", func() {
+			It("won't create a cache file", func() {
+				fakeClock := &fakes.FakeClock{}
+				fakeClock.NowReturns(time.Date(9999, 1, 1, 1, 1, 1, 1, time.UTC))
+				p, err := NewFileCacheProvider("profile", credentials.NewStaticCredentials("id", "secret", "token"), fakeClock)
+				Expect(err).ToNot(HaveOccurred())
+				_, err = p.Retrieve()
+				Expect(err).ToNot(HaveOccurred())
+				_, err = os.Stat(filepath.Join(tmp, "credentials.yaml"))
+				Expect(os.IsNotExist(err)).To(BeTrue())
+			})
+
+		})
+		When("the cache file is not private", func() {
```

Comment:
instead of private (which is hard to tell what it means) can we say what is expected in octal?